src/artm/regularizer/itopic_theta.cc

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014, Additive Regularization of Topic Models.
+
+// Author: me
+
+#include <vector>
+
+#include "glog/logging.h"
+#include "artm/core/protobuf_helpers.h"
+
+#include "artm/regularizer/itopic_theta.h"
+
+iTopicThetaAgent
+iTopicTheta
+
+namespace artm {
+namespace regularizer {
+
+void iTopicThetaAgent::Apply(int item_index, int inner_iter, int topics_size, float* theta) const {
+  auto phi = GetPhiMatrix(instance_->config()->nwt_name());
+  // now we need to get tokens inside current document
+  const Item& item = mybatch.item(item_index);
+  for (int token_index = 0; token_index < item.token_id_size(); ++token_index) {
+    int token_id = item.token_id(token_index);
+    ClassId class_id = batch.class_id(token_id);
+    if (class_id == config_.class_name) {
+      float token_weight = item.token_weight(token_index);
+
+      // NOTE: we could use class_weight * token_weight here instead
+      // but that makes combining iTopicTheta with log-likelihood 
+      // regularizer more tricky for no actual gain
+      for (int topic_id = 0; topic_id < topics_size; ++topic_id) {
+        theta[topic_id] += token_weight * phi->get(token_index, topic_index);
+      }
+    }
+  }
+}
+
+iTopicTheta::iTopicTheta(const iTopicThetaConfig& config) : config_(config) { }
+
+std::shared_ptr<RegularizeThetaAgent>
+iTopicTheta::CreateRegularizeThetaAgent(const Batch& batch,
+                                                const ProcessBatchesArgs& args, double tau) {
+  iTopicThetaAgent* agent = new iTopicThetaAgent();
+  std::shared_ptr<iTopicThetaAgent> retval(agent);
+
+  const int topic_size = args.topic_name_size();
+  const int item_size = batch.item_size();
+  agent->mybatch = batch;
```

Comment:
Does this even compile? I see that ``mybatch`` is a ``const Batch&``, e.g. reference type. I though this must be set in constructor initializer list. I'm quite sure some compilers will enforce this limitation.